pandas/core/dtypes/cast.py

```diff
@@ -357,18 +357,20 @@ def trans(x):
             # if we don't have any elements, just astype it
             return trans(result).astype(dtype)
 
-        # do a test on the first element, if it fails then we are done
         r = result.ravel()
-        arr = np.array([r[0]])
 
-        if isna(arr).any():
-            # if we have any nulls, then we are done
+        if isna(r[0]):
+            # do a test on the first element, if it fails then we are done
```

Comment:
This test on the first element goes back to https://github.com/pandas-dev/pandas/commit/dc73315d557874a7f162dcf0bc3ee106f9338ec5#diff-fb8a9c322624b0777f3ff7e3ef8320d746b15a2a0b80b7cab3dfbe2e12e06daa in `core.common`. It appears to me it is no longer necessary.

Reply:
So I constructed a simple test case and I think I understand what the earlier code was trying to do, which was to `ravel` only the first element of `result`.  Consider:

```
import numpy as	np
import pandas as pd

aa = np.array([1.0, np.nan, 2.0])
bb = np.array([aa, aa])

print(aa, pd.isna(aa))
# [ 1. nan  2.] [False  True False]
print(bb, pd.isna(bb))
#  [[ 1. nan  2.] [ 1. nan  2.]] [[False  True False] [False  True False]]
print(pd.isna(bb).any())
# True
```
The fact that `pd.isna(bb).any()` sees the nulls inside the sub-arrays means we don't need to ravel the whole thing to look for nulls.  The reason to ravel, instead, is to look at the type of the elements, and we only need to ravel the first element to look into that.  (We have already tested that there are elements to ravel.)